libavfilter/dnn/dnn_backend_openvino.c

```diff
@@ -173,18 +189,37 @@ DNNModel *ff_dnn_load_model_ov(const char *model_filename, const char *options)
     if (!ov_model)
         goto err;
     ov_model->ctx.class = &dnn_openvino_class;
+    ctx = &ov_model->ctx;
 
     status = ie_core_create("", &ov_model->core);
     if (status != OK)
         goto err;
 
+    //parse options
+    av_opt_set_defaults(ctx);
+    if (av_opt_set_from_string(ctx, options, NULL, "=", "&") < 0) {
+        av_log(ctx, AV_LOG_ERROR, "Failed to parse options \"%s\"\n", options);
+        goto err;
+    }
+
     status = ie_core_read_network(ov_model->core, model_filename, NULL, &ov_model->network);
     if (status != OK)
         goto err;
 
-    status = ie_core_load_network(ov_model->core, ov_model->network, "CPU", &config, &ov_model->exe_network);
-    if (status != OK)
+    status = ie_core_load_network(ov_model->core, ov_model->network, ov_model->ctx.options.device_type, &config, &ov_model->exe_network);
+    if (status != OK) {
+        status = ie_core_get_available_devices(ov_model->core, &a_dev);
+        if (status != OK) {
+            av_log(ctx, AV_LOG_ERROR, "Failed to get available devices\n");
+            goto err;
+        }
+        for (int i = 0; i < a_dev.num_devices; i++)
+            APPEND_STRING(all_dev_names, a_dev.devices[i])
+        av_log(ctx, AV_LOG_ERROR,
+               "Failed to init OpenVINO model, device %s may not be supported, all available devices are: \"%s\"\n",
```

Comment:
first part of the message should be after line 210.

Reply:
modified, this kind of inappropriate code will be noticed in future